### PR TITLE
[ready to merge] Fixes #84

### LINKED ISSUE #84: pre+code tag instead of just pre
In order to make **PrismJS syntax highlighting** work, it requires all code blocks to be formatted like so:
```
<pre>
  <code>
// the code
  </code>
</pre>
```

But snarkdown only uses `<pre>` without `<code>`.

My request is that snarkdown uses `<pre><code>` by default OR via an option.

From the PrismJS docs:

> Encourages good author practices. Other highlighters encourage or even force you to use elements that are semantically wrong, like `<pre>` (on its own) or `<script>`. Prism forces you to use the correct element for marking up code: `<code>`. On its own for inline code, or inside a `<pre>` for blocks of code. In addition, the language is defined through the way recommended in the HTML5 draft: through a `language-xxxx` class.

https://prismjs.com/#features-full

### PR DESCRIPTION
**Pull request by mesqueeb**

### CODE REVIEW

**Comment by mesqueeb** [comment]
Hi Jason (@developit)

Do you have any opinion on this change? If all is OK, I can update the tests as well to prepare for merge.

Thank you for your time.

**Comment by developit** [comment]
This looks good to me! If you've got the time to add tests that'd be great, though I'm fine merging this since its a fairly straightfoward fix.

**Comment by mesqueeb** [comment]
Hi @developit 
Thanks for giving me a few days to adjust the update the tests as well.

You'll see that all tests now pass again with the new `<pre><code>` syntax.

Also, I've updated `microbundle` to latest version, because building failed otherwise.

And I've also added the `language-${token}` class to the `<pre>` tag.

Looking forward to your deploy! 🎉

**Comment by mesqueeb** [comment]
(nvm that last commit. prismJS needs it to be on the `<pre>` tag to work)

**Comment by mesqueeb** [comment]
@developit it seems that adding a class to `<pre>` makes the build fail because it's too large.
Shall I remove it again?
This is required for prismJS highlighting to work, but is something I could add manually with a `.replace()` as well...

As a matter of fact, it's possible to manually replace even the `<pre>` with `<pre><code>` as well, making this PR useless. 😅

**Comment by developit** [comment]
Hey @mesqueeb - that is for all your work on this. I will check it out locally, but if you know how much the size increased that'd be useful.

**Comment by mesqueeb** [comment]
@developit no problem!

I have a fork with a branch called `dist` here:
https://github.com/mesqueeb/snarkdown/tree/dist

which has the built files of this patch.

I'm not sure which built file I should look at and how to best determine the size:
image

Because I believe the file system might take up more size than the actual code?

**Comment by mesqueeb** [comment]
@developit I was able to get more info from the Github Actions test runner:
image

It seems like its not the size, but just a problem with NPM? (npm was down when I made these commits, perhaps that was related...)

**Comment by mesqueeb** [comment]
@developit I tried deleting `package-lock.json` (against npm's recommendations), but that seems to present another new issue entirely:
image

I'm not sure what tweaks would be required to make this CI pass...

**Comment by mesqueeb** [comment]
I reverted the revert. Because prismJS shows it should be on `<code>` anyway (as per HTML draft recommendation), but they write that the prismJS library will copy it to the `<pre>` tag as well.
image

In any case. Updated so the tests now check it's on `<code>`. And I double checked with PrismJS CDN usage, works like a charm.

If this gets merged and published to NPM I will make a new PR to update the Readme with a short guide on how to enable PrismJS highlighting with CDN and via node.

**Comment by mesqueeb** [comment]
@developit  How've you been? I hope you're doing well!

## TLDR;
- double checked with PrismJS CDN usage, works like a charm. (the intent of this PR)
- If this gets merged, will make a new PR to update the Readme with **a short guide on PrismJS highlighting** (the intent of this PR)
- the only file over 1000bytes is `snarkdown.umd.js.gz`, see image:

image

What's your policy on a single build file over 1000b?

**Comment by mesqueeb** [comment]
@developit Since you already approved this PR, are you feeling OK to work with me towards merging this?

I'm having some issues with dependencies of mine that have to use my fork from github that clash with other dependencies that are using the main package.

**Comment by developit** [comment]
Hi @mesqueeb - sorry about the delay, I'll try to get this dealt with today. I think it's good to go, just need to check a couple things and do the release.
